src/OpenTelemetry.ClrProfiler.Native/startup_hook.h

```diff
@@ -0,0 +1,29 @@
+#ifndef DD_CLR_PROFILER_STARTUP_HOOK_H_
+#define DD_CLR_PROFILER_STARTUP_HOOK_H_
+
+#include "string.h"  // NOLINT
+#include "pal.h"
+
+namespace trace
+{
+
+inline WSTRING GetExpectedStartupHookPath(WSTRING home_path) {
+    if (home_path.back() != DIR_SEPARATOR) {
+        home_path.push_back(DIR_SEPARATOR);
```

Comment:
nit: pass `home_path` as a const reference and put the potential separator, either empty string or DIR_SEPARATOR, avoids the copy. Very nit :) happens only once at initialization...